When building use the libtool intermediate .lo files instead of explicitly using
the .o files. Under libtool foo.lo is the libtool intermediate wrapper, foo.o is
a static build, and .libs/foo.o is a shared build.

If static libraries have been disabled globally then libtool won't generate them
and explicit references to foo.o won't be satisfied.

Upstream-Status: Pending
Signed-off-by: Ross Burton <ross.burton@intel.com>

diff --git a/Makefile b/Makefile
index bb695c6..5f60146 100644
--- a/Makefile
+++ b/Makefile
@@ -28,7 +28,7 @@ endif
 
 .SUFFIXES:
 
-OBJS=apmlib.o
+OBJS=apmlib.lo
 EXES=apm apmd xapm apmsleep
 HEADERS=apm.h
 
@@ -66,22 +66,22 @@ all: $(EXES)
 
 $(OBJS): $(HEADERS)
 
-%.o: %.c
+%.lo: %.c
 	$(LT_COMPILE) -c $(CPPFLAGS) $(CFLAGS) $(XTRACFLAGS) $<
 
-%: %.o $(LIBAPM)
+%: %.lo $(LIBAPM)
 	$(LT_LINK) -o $@ $< $(LDFLAGS) $(LIBAPM)
 
-xapm.o: xapm.c
+xapm.lo: xapm.c
 	$(LT_COMPILE) -c $(CPPFLAGS) $(CFLAGS) $(XTRACFLAGS) -DNARROWPROTO $<
 
-apmd: apmd.o
+apmd: apmd.lo
 
-apmsleep: apmsleep.o
+apmsleep: apmsleep.lo
 
-apmexists: apmexists.o
+apmexists: apmexists.lo
 
-xapm: xapm.o $(LIBAPM)
+xapm: xapm.lo $(LIBAPM)
 	$(LT_LINK) -o $@ $< $(LDFLAGS) $(LIBAPM) $(XLDFLAGS) $(XLIBS)
 
 $(LIBAPM): apmlib.lo
